src/extension.ts

```diff
@@ -63,12 +79,29 @@ export function activate(context: ExtensionContext) {
     });
 
     workspace.onDidSaveTextDocument(doc => {
-        if (ameba && ameba.config.onSave) ameba.execute(doc);
+        if (ameba && doc.languageId === 'crystal' && !doc.isUntitled && doc.uri.scheme === 'file') {
+            outputChannel.appendLine(`[Save] Running ameba on ${getRelativePath(doc)}`)
+            ameba.execute(doc);
+        }
```

Comment:
Please mention this bugfix/change in the PR title/description.

Reply:
btw, is it possible for `doc.uri.scheme` to be anything else then `file`?

Reply:
This is just part of adding logging, moving some of the checks from `ameba.execute()` to this function (so it doesn't log it's running ameba when it's an invalid file).

> is it possible for doc.uri.scheme to be anything else then file?

No idea, I do similar checks for `vscode-crystal-lang` (and `ameba.execute()` already has this check).